### DISCUSSION TITLE
Combine localStorage with Immer

### DISCUSSION THREAD

**Discussion by SteliosKornelakis** [pmndrs/jotai#933] [Q&A] [answered]
How can we combine Immer with Local Storage with Jotai?
I cannot find any working examples.

**Comment by dai-shi** [comment, marked as answer]
Combining utils is still open for research.
For historical reason, we have a solution for immer.
We don't describe it well in docs, because this seems a special case.

```js
import { atomWithStorage } from 'jotai/utils'
import { withImmer } from 'jotai/immer'

const anAtom = withImmer(atomWithStorage('count', 0))
```

**Reply by BennyAlex** [reply]
@dai-shi 
Hey, I get withImmer is marked as deprecated. Is there a better way now to archive this?
Thanks :D

**Reply by dai-shi** [reply]
withImmer is marked as deprecated only because it's moved to another package `jotai-immer`. Just import it from `jotai-immer` instead of `jotai/immer`.

**Reply by BennyAlex** [reply]
@dai-shi 
Do you have a quick idea, how use an user-atom withStorage only when the user clicked rememberMe on the login page? 
So if rememberMe is true, userAtom = atomWithStorage, else just an normal atom (which states is only persistent for this session) ?

**Reply by dai-shi** [reply]
Just don't use `atomWithStoage`. Creating such an own atom is fairly easy. You should be able to learn basic patterns in https://jotai.org/docs/guides/persistence. It will require some more tweaks to handle the "condition". Hope it helps.
